app/src/marketplace/containers/EditProductPage/SharedSecretEditor.jsx

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import React, { useEffect, useCallback, useState } from 'react'
+import cx from 'classnames'
+import { I18n } from 'react-redux-i18n'
+
+import KeyField from '$userpages/components/KeyField'
+import AddKeyField from '$userpages/components/KeyField/AddKeyField'
+import useProduct from '../ProductController/useProduct'
+import { getSecrets, postSecret, putSecret, deleteSecret } from '$mp/modules/communityProduct/services'
+import { isCommunityProduct } from '$mp/utils/product'
+
+import styles from './sharedSecretEditor.pcss'
+
+type Props = {
+    className?: string
+}
+
+type Secret = {
```

Comment:
I'm starting to dislike these Flow definitions but these could moved to the `communityProduct` module or to the `flowtype` definitions.